smack-extensions/src/main/java/org/jivesoftware/smackx/bob/BoBManager.java

```diff
@@ -0,0 +1,134 @@
+import org.jivesoftware.smack.SmackException.NoResponseException;
+import org.jivesoftware.smack.SmackException.NotConnectedException;
+import org.jivesoftware.smack.SmackException.NotLoggedInException;
+import org.jivesoftware.smack.XMPPConnection;
+import org.jivesoftware.smack.XMPPConnectionRegistry;
+import org.jivesoftware.smack.XMPPException.XMPPErrorException;
+import org.jivesoftware.smack.packet.IQ.Type;
+import org.jivesoftware.smackx.bob.element.BoBIQ;
+import org.jivesoftware.smackx.disco.ServiceDiscoveryManager;
+import org.jxmpp.jid.Jid;
+
+/**
+ * Bits of Binary manager class.
+ * 
+ * @author Fernando Ramirez
+ * @see <a href="http://xmpp.org/extensions/xep-0231.html">XEP-0231: Bits of
+ *      Binary</a>
+ */
+public final class BoBManager extends Manager {
+
+    public static final String NAMESPACE = "urn:xmpp:bob";
+
+    static {
+        XMPPConnectionRegistry.addConnectionCreationListener(new ConnectionCreationListener() {
+            @Override
+            public void connectionCreated(XMPPConnection connection) {
+                getInstanceFor(connection);
+            }
+        });
+    }
+
+    private static final Map<XMPPConnection, BoBManager> INSTANCES = new WeakHashMap<>();
+
+    /**
+     * Get the singleton instance of BoBManager.
+     *
+     * @param connection
+     * @return the instance of BoBManager
+     */
+    public static synchronized BoBManager getInstanceFor(XMPPConnection connection) {
+        BoBManager bobManager = INSTANCES.get(connection);
+
+        if (bobManager == null) {
+            bobManager = new BoBManager(connection);
+            INSTANCES.put(connection, bobManager);
+        }
+
+        return bobManager;
+    }
+
+    private BoBManager(XMPPConnection connection) {
+        super(connection);
+    }
+
+    /**
+     * Returns true if Bits of Binary is supported by the server.
+     * 
+     * @return true if Bits of Binary is supported by the server.
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public boolean isSupportedByServer()
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        return ServiceDiscoveryManager.getInstanceFor(connection()).serverSupportsFeature(NAMESPACE);
+    }
+
+    /**
+     * Request BoB data.
+     * 
+     * @param to
+     * @param bobHash
+     * @return the BoB data
+     * @throws NotLoggedInException
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public BoBData requestBoB(Jid to, BoBHash bobHash) throws NotLoggedInException, NoResponseException,
+            XMPPErrorException, NotConnectedException, InterruptedException {
+        BoBIQ requestBoBIQ = new BoBIQ(bobHash);
+        requestBoBIQ.setType(Type.get);
+        requestBoBIQ.setTo(to);
+
```

Comment:
Caching, as described in the XEP, would be nice. But this can be added later on too.